addons/sale_product_configurator/static/src/js/product_configurator_dialog/product_configurator_dialog.js

```diff
@@ -0,0 +1,445 @@
+     * @return {Boolean} - Whether the combination is valid or not.
+     */
+    _isPossibleCombination(product) {
+        return product.attribute_lines.every(ptal => !ptal.attribute_values.find(
+            ptav => ptav.id === ptal.selected_attribute_value_id
+        ).excluded);
+    }
+
+    /**
+     * Check if all the products selected have a valid combination.
+     *
+     * @return {Boolean} - Whether all the products selected have a valid combination or not.
+     */
+    isPossibleConfiguration() {
+        return [...this.state.products].every(
+            p => this._isPossibleCombination(p)
+        );
+    }
+
+    /**
+     * Serialize the product into a format understandable by `sale.order.line`.
+     *
+     * @param {Object} product - The product to serialize.
+     * @return {Object} - The serialized product.
+     */
+    _serializeProduct(product) {
+        let serializedProduct = {
+            product_id: [product.id, product.display_name],
+            product_uom_qty: product.quantity,
+        }
+
+        // handle custom values & no variants
+        let customValuesCommands = [{ operation: "DELETE_ALL" }];
+        for (const ptal of product.attribute_lines) {
+            const selectedCustomPTAV = ptal.attribute_values.find(
+                ptav => ptav.is_custom && ptav.id === ptal.selected_attribute_value_id
+            );
+            if (selectedCustomPTAV) customValuesCommands.push({
+                operation: "CREATE",
+                context: [
+                    {
+                        default_custom_product_template_attribute_value_id: selectedCustomPTAV.id,
+                        default_custom_value: ptal.customValue,
+                    },
+                ],
+            });
+        }
+        serializedProduct.product_custom_attribute_value_ids = {
+            operation: "MULTI",
+            commands: customValuesCommands,
+        };
+
+        let noVariantCommands = [{ operation: "DELETE_ALL" }];
+        const noVariantPTAVIds = product.attribute_lines.filter(
+            ptal => ptal.create_variant === "no_variant"
+        ).map(ptal => { return {id: ptal.selected_attribute_value_id}});
+        if (noVariantPTAVIds.length) noVariantCommands.push({
+            operation: "ADD_M2M",
+            ids: noVariantPTAVIds,
+        });
+        serializedProduct.product_no_variant_attribute_value_ids = {
+            operation: "MULTI",
+            commands: noVariantCommands,
+        };
+
+        return serializedProduct;
+    }
+
+    /**
+     * Confirm the current combination(s).
+     *
+     * @return {undefined}
+     */
+    async onConfirm() {
+        if (!this.isPossibleConfiguration()) return;
+        // Create the products with dynamic attributes
+        for (const product of this.state.products) {
+            if (product.attribute_lines.some(ptal => ptal.create_variant === "dynamic")) {
```

Comment:
```suggestion
            if (!product.id && product.attribute_lines.some(ptal => ptal.create_variant === "dynamic")) {
```
to try (maybe one rpc less in some cases)